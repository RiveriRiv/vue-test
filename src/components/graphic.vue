<template>
	<table><tr v-for="n in temperatures" v-bind:key="n">{{n}}<td v-bind:class="{colored: (n <= temperature && y === day)}" v-for="y in daysNumbers" v-bind:key="y"></td></tr></tr><tr><td></td><td v-for="day in days" v-bind:key="day">{{day}}</td></tr></table>
</template>

<script>
export default {
	data: function() {
    return {
      temperatures: [40, 35, 30, 25, 20, 15, 10, 5, 0],
      daysNumbers: [0, 1, 2, 3, 4, 5, 6],
      days: ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
    }
  },
  computed: {
        town() {
            return this.$store.state.town
        },
        temperature() {
            return this.$store.state.temperature
        },
        day() {
            return this.$store.state.day
        }
    }
	}
</script>