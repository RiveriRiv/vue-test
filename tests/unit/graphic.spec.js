import { createLocalVue } from '@vue/test-utils'
import { mount } from "@vue/test-utils";
import graphic from "@/components/graphic.vue";
import Vuex from 'vuex';

const localVue = createLocalVue()

localVue.use(Vuex)

describe("graphic.vue", () => {
  let actions
  let store
  let state

  beforeEach(() => {
	state = {
      town: 'Moscow',
	  temperature: 20,
	  day: 2
    }

    actions = {
      updateTown: jest.fn(),
      updateTemperature: jest.fn()
    }
    store = new Vuex.Store({
	  state,
      actions
    })
  })

  it("is vue instance", () => {
    const wrapper = mount(graphic, {store, localVue});
    expect(wrapper.isVueInstance()).toBe(true)
  });

  it("matches snapshot", () => {
    const wrapper = mount(graphic, {store, localVue});
     expect(wrapper.html).toMatchSnapshot();
  });

  it("mathes props data", () => {
    const wrapper = mount(graphic, {store, localVue});
 
    expect(wrapper.vm.temperatures).toStrictEqual([40, 35, 30, 25, 20, 15, 10, 5, 0]);
	expect(wrapper.vm.daysNumbers).toStrictEqual([0, 1, 2, 3, 4, 5, 6]);
	expect(wrapper.vm.days).toStrictEqual(['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']);
  });
});
