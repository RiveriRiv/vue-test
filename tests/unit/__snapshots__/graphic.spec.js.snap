// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphic.vue matches spanpshot 1`] = `[Function]`;

exports[`graphic.vue renders props.msg when passed 1`] = `[Function]`;
